src/main/java/thaumicenergistics/common/tiles/TileAdvancedInfusionProvider.java

```diff
@@ -126,22 +126,10 @@ public void grabAllAspects(final TileInfusionMatrix matrix) {
         }
     }
 
-    @Override
-    @TileEvent(TileEventType.WORLD_NBT_WRITE)
-    public void onSaveNBT(final NBTTagCompound data) {
-        super.onSaveNBT(data);
-    }
-
-    @Override
-    @TileEvent(TileEventType.WORLD_NBT_READ)
-    public void onLoadNBT(final NBTTagCompound data) {
-        super.onLoadNBT(data);
-    }
-
     @Override
     public void addWailaInformation(List<String> tooltip) {
         super.addWailaInformation(tooltip);
-        if (this.matrixs == null || (this.matrixs != null && this.matrixs.isEmpty())) {
+        if (this.matrices == null || (this.matrices != null && this.matrices.isEmpty())) {
```

Comment:
Since `matrices` field is always initialized this null check is redundant:
```suggestion
        if (this.matrices.isEmpty()) {
```